### DISCUSSION TITLE
Message localization

### DISCUSSION THREAD

**Discussion by PetSerAl** [dotnet/roslyn#74896] [Q&A] [answered]
Does this host respect `DOTNET_CLI_UI_LANGUAGE` anywhere?
https://github.com/dotnet/roslyn/tree/main/src/Workspaces/Core/MSBuild.BuildHost

I have an issue where messages not properly localized (it use OS language instead of configured language) in VS Code. I can track it to this assembly shipped with `ms-dotnettools.csharp` VS Code extension. I wrote custom MSBuild task, and it seems `CurrentUICulture` is not properly set, when task executed from this host.

**Comment by 333fred** [comment]
@jasonmalinowski or @dibarbet would know.

**Reply by dibarbet** [reply, marked as answer]
Looks like we don't - https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/MSBuild/MSBuild/BuildHostProcessManager.cs#L158

Filed https://github.com/dotnet/roslyn/issues/74902

There is an open question though of whether we should respect the `DOTNET_CLI_UI_LANGUAGE` or the VSCode's configured language.
